bazel/foreign_cc/BUILD

```diff
@@ -64,24 +64,47 @@ envoy_cmake_external(
 
 envoy_cmake_external(
     name = "curl",
+    # Options from https://github.com/curl/curl/blob/master/CMakeLists.txt.
     cache_entries = {
         "BUILD_CURL_EXE": "off",
-        "BUILD_SHARED_LIBS": "off",
-        "CMAKE_BUILD_TYPE": "RelWithDebInfo",
-        "HTTP_ONLY": "on",
         "BUILD_TESTING": "off",
-        "CMAKE_USE_OPENSSL": "off",
-        "CURL_CA_PATH": "none",
+        "BUILD_SHARED_LIBS": "off",
         "CURL_HIDDEN_SYMBOLS": "off",
         "CMAKE_USE_LIBSSH2": "off",
+        "CURL_BROTLI": "off",
+        "CMAKE_USE_GSSAPI": "off",
+        "HTTP_ONLY": "on",
+        "CMAKE_BUILD_TYPE": "RelWithDebInfo",
         "CMAKE_INSTALL_LIBDIR": "lib",
-        "INCLUDE_DIRECTORIES": "include/curl",
+        # C-Ares.
+        "ENABLE_ARES": "on",
```

Comment:
Did you confirm this does what you intended to do? Per https://github.com/curl/curl/blob/curl-7_65_3/docs/INSTALL.cmake c-ares is not supported via CMake. If cmake is incapable here, feel free to switch to configure_make instead. (examples are: https://github.com/bazelbuild/rules_foreign_cc/blob/master/examples/configure_gnuplot/BUILD)

Reply:
I'm not a believer in the up-to-date docs (note how this one is a year old but the CMakeLists has been getting regular updates) so I just used the CMakeLists file itself as the reference, which definitely does support it and the log file confirms `AsyncDNS` is on (CMake out):
```
...
-- Found c-ares: /build/tmp/_bazel_bazel/b570b5ccd0454dc9af9f65ab1833764d/sandbox/processwrapper-sandbox/346/execroot/envoy/bazel_foreign_cc_deps_curl/ares
...
-- Enabled features: IPv6 unix-sockets libz AsynchDNS NTLM HTTP2
...
```

Option here: https://github.com/curl/curl/blob/master/CMakeLists.txt#L78 and here https://github.com/curl/curl/blob/master/CMakeLists.txt#L122

I can also compile a small program with this curl lib to demonstrate it's enabled if that works for you.

Reply:
Cool thanks for confirming, no need to demonstrate 😅  just wanted to double check.

Reply:
I decided to do it just in case to be 100% sure:

```
#include <stdio.h>
#include <curl/curl.h>

int main(int argc, char** argv) {
  curl_version_info_data* info = curl_version_info(CURLVERSION_NOW);
  printf("async DNS: %d\n", !!(info->features & CURL_VERSION_ASYNCHDNS));
  return 0;
}
```

(credit to https://gist.github.com/deltheil/2691bc59a6a6ed03b1cb)

output:

> async DNS: 1

Thanks for the review!

Reply:
@troshko111 Thanks, a bonus point on that: turning it into a unit test to make sure curl and/or other dependencies built correctly with expected options.

Reply:
That's a great idea, sent a PR https://github.com/envoyproxy/envoy/pull/8194.